avm/ptn/network/hub-networking/main.bicep

```diff
@@ -0,0 +1,501 @@
+metadata owner = 'Azure/module-maintainers'
+
+@description('Optional. Location for all Resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+//
+// Add your parameters here
+//
+
+@description('Optional. A map of the hub virtual networks to create.')
+param hubVirtualNetworks hubVirtualNetworkObject
+
+//
+// Add your variables here
+var hubVirtualNetworkPeerings = [for (hub, index) in items(hubVirtualNetworks ?? {}): hub.value.peeringSettings ?? []]
+
+// ============== //
+// Resources      //
+// ============== //
+
+resource avmTelemetry 'Microsoft.Resources/deployments@2023-07-01' = if (enableTelemetry) {
+  name: '46d3xbcp.ptn.network-hubnetworking.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+// Create hub virtual networks
+module hubVirtualNetwork 'br/public:avm/res/network/virtual-network:0.1.6' = [
+  for (hub, index) in items(hubVirtualNetworks ?? {}): {
+    name: '${uniqueString(deployment().name, location)}-${hub.key}-nvn'
+    params: {
+      // Required parameters
+      name: hub.key
+      addressPrefixes: hub.value.addressPrefixes
+      // Non-required parameters
+      ddosProtectionPlanResourceId: hub.value.ddosProtectionPlanResourceId ?? ''
+      diagnosticSettings: hub.value.diagnosticSettings ?? []
+      dnsServers: hub.value.dnsServers ?? []
+      enableTelemetry: hub.value.enableTelemetry ?? true
+      flowTimeoutInMinutes: hub.value.flowTimeoutInMinutes ?? 0
+      location: hub.value.location ?? ''
+      lock: hub.value.lock ?? {}
+      roleAssignments: hub.value.roleAssignments ?? []
+      subnets: hub.value.subnets ?? []
+      tags: hub.value.tags ?? {}
+      vnetEncryption: hub.value.vnetEncryption ?? false
+      vnetEncryptionEnforcement: hub.value.vnetEncryptionEnforcement ?? ''
+    }
+  }
+]
+
+// Create hub virtual network peerings
+resource hubVirtualNetworkPeer_remote 'Microsoft.Network/virtualNetworks@2023-11-01' existing = [
+  for (peer, index) in flatten(hubVirtualNetworkPeerings): {
+    name: peer.remoteVirtualNetworkName
+  }
+]
+
+resource hubVirtualNetworkPeer_local 'Microsoft.Network/virtualNetworks@2023-11-01' existing = [
+  for (hub, index) in items(hubVirtualNetworks ?? {}): if (hub.value.enablePeering) {
+    name: hub.key
+  }
+]
+
+resource hubVirtualNetworkPeering 'Microsoft.Network/virtualNetworks/virtualNetworkPeerings@2023-11-01' = [
```

Comment:
Not sure if it helps, but I wanted to call out that the VNET module can also deploy peerings & reverse peerings :)

Reply:
Will be worked on in a future version...